### PR TITLE
Optimized JNI linkage on X86-64

### PR DESCRIPTION
**Pull request by 0dvictor**
Reduced code path and potential register usage.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp [hunk 1]
```diff
@@ -521,40 +521,24 @@ void TR::AMD64JNILinkage::buildJNICallOutFrame(
...
-         scratchReg = cg()->allocateRegister();
-
-      generateRegImm64Instruction(MOV8RegImm64, callNode, scratchReg, tagBits, cg());
-      generateMemRegInstruction(S8MemReg, callNode, tempMR, scratchReg, cg());
-      }
+   generateImmInstruction(PUSHImm4, callNode, tagBits, cg());
 
    // (skip savedPC slot).
    //
+   generateImmInstruction(PUSHImm4, callNode, 0, cg());
```
Comment (cathyzhyi) [inline runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp]: what is the `push 0` for?

Comment (0dvictor) [reply runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp]: The linkage requires a slot for `savedPC`, but its content is ignored; therefore, I put this push instruction with an arbitrary value 0.

**Comment by cathyzhyi** [comment]
Placing stores with faster and shorter pushes. LGTM.
@vijaysun-omr could you merge this one as well?

**Comment by andrewcraik** [comment]
Jenkins test sanity xlinux,win jdk8,jdk11

**Pull request merged**